modules/bioinspired/src/retina_ocl.cpp

```diff
@@ -472,6 +439,13 @@ void RetinaOCLImpl::activateContoursProcessing(const bool activate)
     _retinaFilter->activateContoursProcessing(activate);
 }
 
+// unimplemented interfaces:
+void RetinaOCLImpl::applyFastToneMapping(InputArray /*inputImage*/, OutputArray /*outputToneMappedImage*/) { NOT_IMPLEMENTED; }
+void RetinaOCLImpl::getParvoRAW(OutputArray /*retinaOutput_parvo*/) { NOT_IMPLEMENTED; }
+void RetinaOCLImpl::getMagnoRAW(OutputArray /*retinaOutput_magno*/) { NOT_IMPLEMENTED; }
+const Mat RetinaOCLImpl::getMagnoRAW() const { NOT_IMPLEMENTED; return Mat(); }
+const Mat RetinaOCLImpl::getParvoRAW() const { NOT_IMPLEMENTED; return Mat(); }
```

Comment:
Transparent API must avoid these failures and automatically drop to CPU implementation.